--- ppp-2.4.6.orig/pppd/main.c
+++ ppp-2.4.6/pppd/main.c
@@ -740,8 +733,31 @@ void
 set_ifunit(iskey)
     int iskey;
 {
-    info("Using interface %s%d", PPP_DRV_NAME, ifunit);
-    slprintf(ifname, sizeof(ifname), "%s%d", PPP_DRV_NAME, ifunit);
+    if (req_ifname[0] != '\0') {
+        char *ptr;
+        ptr = strchr(req_ifname, 'N');
+
+        if (ptr == NULL) {
+            slprintf(ifname, sizeof(ifname), "%s", req_ifname);
+            info("Using ifname as it is: '%s'", ifname);
+        } else {
+            int index=0;
+            char req_ifname_t[MAXIFNAMELEN];
+
+            index = (int)(ptr - req_ifname);
+            memset(req_ifname_t, 0, MAXIFNAMELEN); // force cleanup
+            strncpy(req_ifname_t, req_ifname, index);
+
+            // Copy the part of the req_ifname which doesn't contains "N"
+            strncpy(req_ifname_t, req_ifname, index);
+
+            slprintf(ifname, sizeof(ifname), "%s%d", req_ifname_t, ifunit);
+            info("Using ifname in format ifnameN. Set to new name: '%s'", ifname);
+        }
+    }
+    else
+	slprintf(ifname, sizeof(ifname), "%s%d", PPP_DRV_NAME, ifunit);
+    info("Using interface %s", ifname);
     script_setenv("IFNAME", ifname, iskey);
     if (iskey) {
 	create_pidfile(getpid());	/* write pid to file */
